pipeline/matching/generators/similar_interests_test.py

```diff
@@ -40,9 +41,163 @@ def test_example():
             ),
         ),
     ]
-    # TODO: Uncomment this assertion and delete the empty list assert
-    # assert actual == expected
-    assert actual == []
+    assert actual == expected
 
 
-# TODO: Add more test cases for your logic
+# test where users do not have any similar interest
+def test_no_similar_interest():
+    hiking = Interest(code="hiking", name="Hiking")
+    swimming = Interest(code="swimming", name="Swimming")
+    reading = Interest(code="reading", name="Reading")
+    painting = Interest(code="painting", name="Painting")
+
+    inp = MatchingInput(
+        community="test",
+        release="2022-06-26",
+        users=[
+            User(uid="1", displayName="A", interests=[hiking, swimming]),
+            User(uid="2", displayName="B", interests=[painting, reading]),
+        ],
+        recent_matches=[],
+        interests=[hiking, swimming, painting, reading],
+    )
+    generator = SimilarInterestsGenerator(min_common=1)
+    actual = list(generator.generate(inp))
+
+    expected = [
+        Match(
+            users={"1", "2"},
+            metadata=MatchMetadata(
+                generator="similarInterestsGenerator",
+                score=(0.0),
+                commonInterests=[],
+            ),
+        ),
+    ]
+    assert actual == expected
+
+
+# test where one user has more interest than another user
+def test_one_with_many_interest():
+    hiking = Interest(code="hiking", name="Hiking")
+    swimming = Interest(code="swimming", name="Swimming")
+    reading = Interest(code="reading", name="Reading")
+    painting = Interest(code="painting", name="Painting")
+    crocheting = Interest(code="crocheting", name="Crocheting")
+    boxing = Interest(code="boxing", name="Boxing")
+
+    inp = MatchingInput(
+        community="test",
+        release="2022-06-26",
+        users=[
+            User(
+                uid="1",
+                displayName="A",
+                interests=[hiking, swimming, painting, crocheting, boxing],
+            ),
+            User(uid="2", displayName="B", interests=[painting]),
+        ],
+        recent_matches=[],
+        interests=[hiking, swimming, painting, crocheting, boxing],
+    )
+    generator = SimilarInterestsGenerator(min_common=1)
+    actual = list(generator.generate(inp))
+
+    expected = [
+        Match(
+            users={"1", "2"},
+            metadata=MatchMetadata(
+                generator="similarInterestsGenerator",
+                score=(1 / 5),
```

Comment:
Nice test case! Good idea to keep this as a fraction instead of the raw float, makes testing more reliable.